src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1796,17 +1797,75 @@ impl Cage {
         }
     }
 
-    //------------------------------------WRITE SYSCALL------------------------------------
-
+    /// ## ------------------WRITE SYSCALL------------------
+    /// ### Description
+    ///
+    /// The `write_syscall()` attempts to write `count` bytes from the buffer
+    /// pointed to by `buf` to the file associated with the open file
+    /// descriptor, `fd`. The number of bytes written may be less than count
+    /// if, for example, there is insufficient space on the underlying
```

Comment:
Are there another cases where the no of bytes written are less than `count` - perhaps iff offset + count > filesize ? If so adding that would be great here

Reply:
The other case when the no of bytes written are less than count is when this syscall gets interrupted by a signal, which we have not implemented yet. 
For the case you have mentioned (count + offset > filesize), our internal implementation extends the file size and we can write the intended bytes into the file.